### PR TITLE
fix(operator): add OwnerReference for controller

### LINKED ISSUE #1638: bug: kubearmor-operator helm chart doesn't remove kubearmor-controller
## Bug Report

**General Information**

- Environment description: k3s or any k8s
- Kernel version: N/A
- Orchestration system version in use: `v1.23.9+k3s1`
- Link to relevant artifacts: kuebarmor-operator helm charts
- Target containers/pods: kubearmor-controller

**To Reproduce**

1. Install KubeArmor using kubearmor-operator helm chart.
2. Uninstall the helm release.
3. You'll see that kubearmor-controller pod stays installed and goes into error.

**Reason**
The secret containing the cert for mutating admission webhook, mounted as a volume on the controller, is deleted before kubearmor-controller deployment is.

**Expected behavior**

Cleanup should be ordered.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by rootxrishabh**
**Purpose of PR?**:
Earlier controller deployment uninstallation was failing during helm uninstallation due to a lack of ownership of the controller. This PR adds the operator as the owner of the controller deployment

Fixes #1638 

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [X] Bug fix. Fixes #1638 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by rootxrishabh** [comment]
I have tested the functionality and controller uninstalls successfully.
image

**Review by daemon1024 — DISMISSED** [review]

File: pkg/KubeArmorOperator/internal/controller/resources.go [hunk 1]
```diff
@@ -460,6 +460,7 @@ func (clusterWatcher *ClusterWatcher) WatchRequiredResources() {
 	relayServer.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullPolicy(common.KubeArmorRelayImagePullPolicy)
 	deploys := []*appsv1.Deployment{
 		addOwnership(relayServer).(*appsv1.Deployment),
+		addOwnership(controller).(*appsv1.Deployment),
```
Comment (rksharma95) [inline pkg/KubeArmorOperator/internal/controller/resources.go]: let's not add controller deployment to `deploys`, controller deployment need to handled separately here:
https://github.com/kubearmor/KubeArmor/blob/b86dbcc76f94a5520f717331f9e490091c32ebbe/pkg/KubeArmorOperator/internal/controller/resources.go#L347

**Review by daemon1024 — APPROVED** [review]

**Comment by rootxrishabh** [comment]
The failing test seems to be flakey. It passed with a few retries showing the same error in the last push.

**Review by rksharma95 — APPROVED** [review]

**Comment by rksharma95** [comment]
@rootxrishabh can you please rebase the PR. thanks

**Pull request merged**